### ISSUE TITLE
[no-unused-properties] can't recogonize props.xxx used in template

### ISSUE THREAD

**Issue by tjx666** [vuejs/eslint-plugin-vue#2193]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.41.0
- **eslint-plugin-vue version:** 9.14.0
- **Node version:** 18.16.0
- **Operating System:** MacOS

**Please show your full configuration:**

```json5
{
    "rules": {
        'vue/no-unused-properties': 'error',
    }
}
```

**What did you do?**	

```vue
<template>
    <div>
        {{ props.age }}
    </div>
</template>

<script lang="ts" setup>
const props = defineProps<{
    // will show vue/no-unused-properties error
    age: number;
}>();
</script>
```

**What did you expect to happen?**

No errors.

**What actually happened?**

report vue/no-unused-properties error

**Repository to reproduce this issue**

[playground link](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJx9V8ty5DYM/JUpnZIqM67K0et15RP2A3yhKEjDmCIVkppHXPPvaT4kkZqZ+LIrAILARqPB+W6E6ah5a949jZPinj4+9QF/75085f+Gv+/vw2TN5P7gAx1utxzzmoLeX4uX8eiElZM/KK6Hn5+Nd5/NwZGfJ3iF0c6nVIefh456qelXeHr/TjmR/+2g57El++NT3z5++/1H+EBK+dG8NHZW5Jq37yamsrPwxjI3T2Sbtz9fmh5PnbQkvDQ6WgbyniyzqMAmizaMu6sWDHWM0hGjC4kZeRav4DB2JBS3PORxq0Nx5/Cyk8OaSphx4paYpoH9S9Zsdt3dhaLkRzaufYyXa9XJ461RqHygy2LsqJ2Hgez2rMgTO/HNAiwYt4OrDKnykQKwtYcUDiv7yvZF1zJISYHmRlgWKxrur9UDE0fAJQLW8WO1c+KhC+vZ6LKDgS7ectYao4jr8CVfexyNcrH0XCl/tGYejqtpRjvrjIMyLVc7oxwnY/ctkFqj6Ef9lvrElexSC6bVavE8I5idj9KTm7hYcVEGMJse1CIhXdFNEA1H66QengE1ygt1LGYDx0Aez9vVqenM3HVsjVotRhvhcUIULkf8S07waS3EtH8DRqXWDDF4eQDzvfFXtLqdpfJyO7KlBARtBpw9l7XYHKl+B6J7NGbuyDtzhhGTKnHijmk+FmkwOC6Q1fKNb/4oHWsJg0zFXMMxa+jF9oCGUEw5zjiAgpBsPkscILeqMDneE4PcAJLrzorJ5eXgZbOZghHAIpXUaNzmnx2+q3hLG7zZiEHcTICJQIeWiy9LPVnSG1EWJyZLHPfGupVn6VOEpX9maBu7SlJdtMxhdp3mqfZE1tBWk4A6zfTaKiO+WBDjncnzIdBqRS54gpgZTZAjPknm/FU98oUmYjbYIvthXmWRf4tMGLr0xj5qdt6MjE5ryl1AWg8QEOldOZ77gJELazB1tstkKXxx1Tx8+ehHBfZ7bzQ7chdhq504xkhJmX2scQfWw6AwKOUhqiAJCmu/+sbQe7ah1UtFEYmnEVm/qpggDaEuDI0/E+k0EMkUjk82K3UKvnjmJdpfNxc0a8MWC4OqB1d2t4rJAg10Ob5SurY9Aam3SHAkK7HZRJUgr4r1QJGKZUAsfgIOEDDMd6xkk8kcNJmAteSK7akW2mjKUAxeToWi04VhtlRHxKnqIlhx2B4dr9Cwu5oLX5BcxvsghfzM5bMUdycqffvpeRZXzM+TkEg+SEog37M0cUKe+HAzAVdiQyVGhZ6EnVgLZldOYzzMsi+NOZvUkZqRf9WHVznxuL5Q4AbXFc5xNYAPeTvgBoBgHK0Y+2VPbCi6e1+eCkk7H9ZNGC+cqCceaLLzR4V/9nJ0gm21OQv6iF5hJdFD5z18J2Bx2Zo68S5eHKopjyx0/x8DiX8ckcZqJVoUS1nJ4xQ3Fssyumdi9saJSdeILAy9tM7vwxxNQX1pYUDN/hyE0aSNabg6oBNHXuCybL90vce9MPTqzhu2BYuLkC83+sqNtxzdmeP6uRPLxRuPmTV8daafJHtJd0Ge1+tztBRnTtef3aY6sfijhZRE6QGtSphPLO4n3al7V1z3dx26vTTxXmXDr6RA/+av0KJcLzn037/miNvtP3SK5iU=)

**Comment by FloEdelmann** [comment]
Related:

* #2344
* #1285
